### PR TITLE
be lenient and retry 5 times when to give limited tables time to settle

### PR DESCRIPTION
**Pull request by jwiebema1**

### CODE REVIEW

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java [hunk 1]
```diff
@@ -163,35 +163,51 @@ public void run() {
...
         VoltTable partitions = client.callProcedure("@GetPartitionKeys",
                 "INTEGER").getResults()[0];
-        long count = TxnId2Utils.doAdHoc(client,"SELECT COUNT(*) FROM capr;").getResults()[0].fetchRow(0).getLong(0);
-        if (count > 10) {
-            // retry once
-            log.warn("CappedTableLoader Replicated table CAPR has more rows ("+count+") than the limit set by capped collections (10) retrying in 1 sec");
-            try { Thread.sleep(1000); } catch (Exception e2) {}
+        int retries = 5;
+        boolean lastSuccess = false;
+        long count = 100000;
```
Comment (jmcgrail-voltdb) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java:168]: perhaps Long.MIN_VALUE?

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java [hunk 2]
```diff
@@ -163,35 +163,51 @@ public void run() {
...
+        while ( retries > 0 ) {
             count = TxnId2Utils.doAdHoc(client,"SELECT COUNT(*) FROM capr;").getResults()[0].fetchRow(0).getLong(0);
-            if ( count > 10 ) {
-                log.error("CappedTableLoader Replicated table CAPR has more rows ("+count+") than the limit set by capped collections (10)");
-                return(true);
+            if (count > (10*1.15)) {
```
Comment (jmcgrail-voltdb) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java:171]: a comment for why the "*.1.15" is included?

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java [hunk 3]
```diff
@@ -163,35 +163,51 @@ public void run() {
...
-                try { Thread.sleep(1000); } catch (Exception e2) {}
+            retries = 5;
+            count = 10000;
+            lastSuccess = false;
+            while ( retries > 0 ) {
                 count = client.callProcedure("CAPPCountPartitionRows",key).getResults()[0].fetchRow(0).getLong(0);
-                if (count > 10) {
-                    log.error("CappedTableLoader Partitioned table CAPP has more rows ("+count+") than the limit set by capped collections (10) on partition "+id);
-                    return(true);
+                // give it 15% lee-way
```
Comment (jmcgrail-voltdb) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java:195]: aah, i see here a comment for the 1.15 :)

Perhaps a function if these two loops are the same?

Comment (jwiebema1) [reply tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/CappedTableLoader.java:195]: One of the loops is for a replicated table and the other is for partitioned table and needs the partition info so they are different.

**Pull request merged**